### PR TITLE
Fixes various flaws in registry signing

### PR DESCRIPTION
**Pull request by jcspencer**
Changes:
- gzipped registry is signed rather than the unzipped registry, meaning that it can be verified without being unzipped
- removes redundant double-hashing

### CODE REVIEW

File: lib/hex_web/registry_builder.ex [hunk 1]
```diff
@@ -109,12 +109,12 @@ defmodule HexWeb.RegistryBuilder do
     :ok = :ets.tab2file(tid, String.to_char_list(file))
     :ets.delete(tid)
 
-    output = File.read!(file)
+
```
Comment (whatyouhide) [inline lib/hex_web/registry_builder.ex]: :scissors:

File: web/utils.ex [hunk 2]
```diff
@@ -222,10 +222,18 @@ defmodule HexWeb.Utils do
     [entry | _ ] = :public_key.pem_decode(key)
     key = :public_key.pem_entry_decode(entry)
 
-    :public_key.sign({:digest, file}, :sha512, key)
+    :public_key.sign(file, :sha512, key)
     |> Base.encode16(case: :lower)
   end
 
+  def verify(file, signature, key) do
+    [entry | _ ] = :public_key.pem_decode(key)
```
Comment (whatyouhide) [inline web/utils.ex]: Extra space before the `]`.

**Comment by jcspencer** [comment]
@whatyouhide fixed and squashed :+1:

**Comment by ericmj** [comment]
Looks good, thanks.

**Pull request merged**